### PR TITLE
Rework DtmiConventions in models repository SDK

### PR DESCRIPTION
**Pull request by digimaun**
Highlights

- Reworked `DtmiToQualifiedPath`  to revolve around Uri. Changed name to `GetModelUri`.
- Model FileFetcher will use LocalPath from the Uri object.
- Update of tests & samples.

### CODE REVIEW

**Review by azabbasi — APPROVED** [review]

**Comment by digimaun** [comment]
cc: @cartertinney

File: sdk/modelsrepository/Azure.Iot.ModelsRepository/src/DtmiConventions.cs [hunk 1]
```diff
@@ -43,24 +43,20 @@ public static string DtmiToQualifiedPath(string dtmi, string basePath, bool from
...
-                basePath += "/";
+                dtmiPath = dtmiPath.Replace(
+                    ModelsRepositoryConstants.JsonFileExtension,
+                    ModelsRepositoryConstants.ExpandedJsonFileExtension);
             }
 
-            string fullyQualifiedPath = $"{basePath}{dtmiPath}";
-
-            if (fromExpanded)
+            UriBuilder repositoryUriBuilder = new UriBuilder(repositoryUri);
```
Comment (drwill-ms) [inline sdk/modelsrepository/Azure.Iot.ModelsRepository/src/DtmiConventions.cs:53]: var

File: sdk/modelsrepository/Azure.Iot.ModelsRepository/src/DtmiConventions.cs [hunk 2]
```diff
@@ -43,24 +43,20 @@ public static string DtmiToQualifiedPath(string dtmi, string basePath, bool from
...
+            if (!repositoryUriBuilder.Path.EndsWith("/", StringComparison.InvariantCultureIgnoreCase))
```
Comment (drwill-ms) [inline sdk/modelsrepository/Azure.Iot.ModelsRepository/src/DtmiConventions.cs:54]: Probably should add a code comment as to why we want the URI to end in a slash.

Comment (digimaun) [reply sdk/modelsrepository/Azure.Iot.ModelsRepository/src/DtmiConventions.cs:54]: Added in new PR https://github.com/Azure/azure-sdk-for-net/pull/19429

**Pull request merged**